druid/src/test/java/org/apache/calcite/test/DruidAdapterIT.java

```diff
@@ -2155,6 +2155,310 @@ public RelNode apply(RelBuilder b) {
+            "store_state=WA; A=Infinity");
+  }
+
+  @Test public void testDivideByZeroDoubleTypeNegInfinity() {
+    final String sqlQuery = "select \"store_state\", -1.0 * sum(\"store_cost\") / 0 as "
+        + "a from \"foodmart\"  group by \"store_state\" order by a desc";
+    String postAggString = "'postAggregations':[{'type':'arithmetic','name':'postagg#0',"
+        + "'fn':'quotient','fields':[{'type':'arithmetic','name':'',"
+        + "'fn':'*','fields':[{'type':'constant','name':'','value':-1.0},"
+        + "{'type':'fieldAccess','name':'','fieldName':'$f1'}]},"
+        + "{'type':'constant','name':'','value':0.0}]}]";
+    final String plan = "PLAN=EnumerableInterpreter\n"
+        + "  DruidQuery(table=[[foodmart, foodmart]], "
+        + "intervals=[[1900-01-09T00:00:00.000/2992-01-10T00:00:00.000]], "
+        + "groups=[{63}], aggs=[[SUM($91)]], projects=[[$0, /(*(-1.0, $1), 0)]]";
+    sql(sqlQuery, FOODMART)
+        .explainContains(plan)
+        .queryContains(druidChecker(postAggString))
+        .returnsOrdered("store_state=CA; A=-Infinity",
+            "store_state=OR; A=-Infinity",
+            "store_state=WA; A=-Infinity");
+  }
+
+  @Test public void testDivideByZeroDoubleTypeNaN() {
+    final String sqlQuery = "select \"store_state\", (sum(\"store_cost\") - sum(\"store_cost\")) "
+        + "/ 0 as a from \"foodmart\"  group by \"store_state\" order by a desc";
+    String postAggString = "'postAggregations':[{'type':'arithmetic','name':'postagg#0',"
+        + "'fn':'quotient','fields':[{'type':'arithmetic','name':'','fn':'-',"
+        + "'fields':[{'type':'fieldAccess','name':'','fieldName':'$f1'},"
+        + "{'type':'fieldAccess','name':'','fieldName':'$f1'}]},"
+        + "{'type':'constant','name':'','value':0.0}]}]";
+    final String plan = "PLAN=EnumerableInterpreter\n"
+        + "  DruidQuery(table=[[foodmart, foodmart]], "
+        + "intervals=[[1900-01-09T00:00:00.000/2992-01-10T00:00:00.000]], "
+        + "groups=[{63}], aggs=[[SUM($91)]], projects=[[$0, /(-($1, $1), 0)]], "
+        + "sort0=[1], dir0=[DESC]";
+    sql(sqlQuery, FOODMART)
+        .explainContains(plan)
+        .queryContains(druidChecker(postAggString))
+        .returnsOrdered("store_state=CA; A=NaN",
+            "store_state=OR; A=NaN",
+            "store_state=WA; A=NaN");
+
+
+  }
+
+  @Test public void testDivideByZeroIntegerType() {
+    final String sqlQuery = "select \"store_state\", (count(*) - "
+        + "count(*)) / 0 as a from \"foodmart\"  group by \"store_state\" "
+        + "order by a desc";
+    final String plan = "PLAN=EnumerableInterpreter\n"
+        + "  DruidQuery(table=[[foodmart, foodmart]], "
+        + "intervals=[[1900-01-09T00:00:00.000/2992-01-10T00:00:00.000]], "
+        + "groups=[{63}], aggs=[[COUNT()]], projects=[[$0, /(-($1, $1), 0)]]";
+    sql(sqlQuery, FOODMART)
+        .explainContains(plan)
+        .throws_("/ by zero");
+  }
+
   /**
```

Comment:
Could you add tests where you interleave aggregate calls between grouping columns in the SELECT clause? And combine that with OrderBy on dimensions for some tests and metrics on some others. This will be needed to test new code introduced for checking code that pushes down Sort operator.